apps/chat/src/components/Marketplace/ApplicationDetails/ApplicationFooter.tsx

```diff
@@ -138,145 +159,320 @@ export const ApplicationDetailsFooter = ({
-                <button
-                  onClick={() => onBookmarkClick(entity)}
-                  className="icon-button"
-                  data-qa="application-bookmark"
+          {isSmallScreen && menuItemsCount > 2 ? (
+            <button className="icon-button">
+              <ContextMenu
+                menuItems={menuItems}
+                featureType={FeatureType.Application}
+                triggerIconHighlight
+                className="m-0 xl:invisible group-hover:xl:visible"
+              />
+            </button>
+          ) : (
+            <>
+              {isPublicApp && isSmallScreen && (
+                <ApplicationCopyLink
+                  reference={entity.reference}
+                  size={24}
+                  hasTooltip
+                  className="icon-button !p-[5px]"
+                />
+              )}
+              {isExecutable && isCodeAppsEnabled && (
+                <Tooltip tooltip={t(getFunctionTooltip(entity))}>
+                  <button
+                    disabled={playerStatus === SimpleApplicationStatus.UPDATING}
+                    onClick={handleUpdateFunctionStatus}
+                    className={classNames(
+                      'icon-button',
+                      PlayerContextButtonClasses[playerStatus],
+                    )}
+                    data-qa="application-status-toggler"
+                  >
+                    <PlayerIcon size={24} />
+                  </button>
+                </Tooltip>
+              )}
+              {isMyApp && isApplicationsSharingEnabled && isSmallScreen && (
+                <Tooltip tooltip={t('Share')}>
+                  <button
+                    onClick={handleOpenSharing}
+                    className="icon-button"
+                    data-qa="application-share"
+                  >
+                    <IconUserShare size={24} />
+                  </button>
+                </Tooltip>
+              )}
+              {!!entity.sharedWithMe && isApplicationsSharingEnabled && (
+                <Tooltip tooltip={t('Unshare application')}>
+                  <button
+                    onClick={handleOpenUnshare}
+                    className="icon-button"
+                    data-qa="application-unshare"
+                  >
+                    <IconUserUnshare height={24} width={24} />
+                  </button>
+                </Tooltip>
+              )}
+              {isMyApp && (
+                <Tooltip tooltip={t(getDisabledTooltip(entity, 'Delete'))}>
+                  <button
+                    disabled={isModifyDisabled}
+                    onClick={() => onDelete(entity)}
+                    className="icon-button"
+                    data-qa="application-delete"
+                  >
+                    <IconTrashX size={24} />
+                  </button>
+                </Tooltip>
+              )}
+              {isApplicationId(entity.id) && (isMyApp || isPublicEntity) && (
```

Comment:
`isMyApp` and `isPublicEntity` already check it
```suggestion
              {(isMyApp || isPublicEntity) && (
```